src/lib/core/popen.c

```diff
@@ -1185,7 +1185,11 @@ popen_new(struct popen_opts *opts)
 	 * posix_spawn here, but only on macOS.
 	 */
 #pragma GCC diagnostic warning "-Wdeprecated-declarations"
+#ifdef TARGET_OS_DARWIN
+	handle->pid = fork();
+#else
 	handle->pid = vfork();
+#endif
```

Comment:
Nope, it does not work. You can enable popen tests back (see d011dffd6e7f768656357afadbc61e10fb3d4f81) and ensure. `fork()` does not block the parent process until `exec*()` in the child process, but the code leans on it.

BTW, the task (#6674) is in my backlog for this quarter, so you can just skip it.

Reply:
Update: The Mac OS problem will be solved in #7658.